pkg/compose/watch.go

```diff
@@ -301,19 +316,24 @@ func (s *composeService) makeSyncFn(
 	project *types.Project,
 	needSync <-chan sync.PathMapping,
 ) func() error {
-	syncer := sync.NewDockerCopy(project.Name, s, s.stdinfo())
+	var syncer sync.Syncer
+	if useTar, _ := strconv.ParseBool(os.Getenv("COMPOSE_EXPERIMENTAL_WATCH_TAR")); useTar {
+		syncer = sync.NewTar(project.Name, tarDockerClient{s: s})
```

Comment:
Yeah, for now I was doing this mostly so that I could incrementally add this and have an E2E run against both implementations while working.

Once things are merged in and look good, I think it might make sense to flip the default to enabled for one release so that `watch` users automatically get the updated experience but can switch back temporarily if needed.

Reply:
Long-term, I think the `docker cp` approach is "worse" in every scenario – the overhead of all the API calls is unavoidable (there's nothing actually wrong with the implementation). So I don't think we need to keep it.

But yes, considering we've discussed e.g. potential for daemon or alternative watch/sync implementations, it's probably worth thinking through what that might look like.